okhttp-tests/src/test/java/okhttp3/HeadersTest.java

```diff
@@ -366,8 +370,14 @@
         .build();
     List<Challenge> challenges = HttpHeaders.parseChallenges(headers, "WWW-Authenticate");
     assertEquals(1, challenges.size());
-    assertEquals("Digest", challenges.get(0).scheme());
+    assertEquals("digest", challenges.get(0).scheme());
     assertEquals("myrealm", challenges.get(0).realm());
+    Map<String, String> expectedAuthParams = new HashMap<>();
+    expectedAuthParams.put("realm", "myrealm");
+    expectedAuthParams.put("nonce", "fjalskdflwejrlaskdfjlaskdjflaksjdflkasdf");
+    expectedAuthParams.put("qop", "auth");
+    expectedAuthParams.put("stale", "FALSE");
+    assertEquals(expectedAuthParams, challenges.get(0).authParams());
 
     // Not strict RFC 2617 header.
```

Comment:
... for what it’s worth, this really wants to be cut into several tests

Reply:
Ok, I split the tests